### DISCUSSION TITLE
After saving the annotation for the second time, the previously saved annotation disappears

### DISCUSSION THREAD

**Discussion by vvvvimy** [sumatrapdfreader/sumatrapdf#4448] [Q&A] [answered]
After saving the annotation for the second time, the previously saved annotation disappears。

**Comment by vvvvimy** [comment]
v3.5.2-64

**Reply by GitHubRulesOK** [reply, marked as answer]
there are 2 primary ways annotations are added to to an existing PDF and both have oddities that can cause losses or corruption.
SumatraPDF uses the safer of the two, namely it should not alter the source PDF but just append the annotation onto the file trailer.

This means it adds incremental pointers each time the file is saved and most likely there was a glitch in that chain of pointers. 
There was a common issue with some ways the pointers worked in MuPDF coding (as modified by SumatraPDF) at one time that should be fixed in more recent 3.6 pre-release.

Latest pre-releases can be found at [https://www.sumatrapdfreader.org/prerelease](https://www.sumatrapdfreader.org/prerelease)

**Reply by vvvvimy** [reply]
okay thank you thank you
